### ISSUE TITLE
JWT Claims are not included in context vars in subsequent requests

### ISSUE THREAD

**Issue by lonelycode** [TykTechnologies/tyk#411]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
When accessing an API with a JWT, and context variables enabled, the context variables are only set the first time the JWT is received.

**What is the expected behavior?**
The JWT claims should always be processed and exposed.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**
1. Create a JWT enabled API
2. Add a header injector that includes one of the claims using the `$tyk_context.jwt_claims_*` namespace
3. Check the request headers after making 2 requests

**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
v2.3.x

**Comment by mvdan** [comment]
Is this a regression?

**Comment by ammaskartik** [comment]
Dont mind attempting the fix, looking at the middleware code, looks like the setContextVars are only set if the session doesnt exist, which makes sense on the issue. Should be a simple fix, good for my first one =)

**Comment by ammaskartik** [comment]
Also, we are kinda blocked by this issue, and would rather not do a workaround.

**Comment by buger** [comment]
@ammaskartik cool! Will wait for your PR then.

**Comment by ammaskartik** [comment]
submitted new pull request: https://github.com/TykTechnologies/tyk/pull/413

**Comment by ammaskartik** [comment]
Looks like @lonelycode merged my pull request and your team is pushing for release v2.3.2, this is great and very much appreciated.

**Comment by ammaskartik** [comment]
@lonelycode whats your policy on closing issues, etc... Only after released?

**Comment by buger** [comment]
@ammaskartik  2.3.2 is a patch release, and have own branch now. You pull request needs to be manually merged into 2.3.2 branch (we will do it, should happen soon), that's why issue is not yet closed :)

Hope it makes sense! Thank you for the contribution!

**Comment by ammaskartik** [comment]
@buger that makes perfect sense. 

Thanks.
